app/models/spree/product.rb

```diff
@@ -474,5 +475,12 @@ def sanitize_permalink
       requested = permalink.presence || permalink_was.presence || name.presence || 'product'
       self.permalink = create_unique_permalink(requested.parameterize)
     end
+
+    def validate_image_for_master
+      return unless master.images.any? { |image| !image.valid? }
```

Comment:
I think this can be slightly simpler without the  negatives:

```suggestion
      return if master.images.all?(&:valid?)
```

Reply:
Thanks for the review, I have made the above changes.